internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -241,6 +252,52 @@ func (cf CloudFormation) Template(stackName string) (string, error) {
 	return cf.cfnClient.TemplateBody(stackName)
 }
 
+// WaitForSignalAndHandleInterrupt waits for a SIGINT signal and handles interrupt while monitoring a CloudFormation stack.
+// It cancels the operation and takes appropriate actions based on the stack status when a signal is received.
+// Returns stack deletion status, update cancellation status, and an error, if any.
+func (cf CloudFormation) WaitForSignalAndHandleInterrupt(ctx context.Context, cancel context.CancelFunc, stackName string) (isDeleted bool, isUpdateCanceled bool, err error) {
+	for {
+		select {
+		case <-cf.signalClient.NotifySignals():
+			cancel()
+			cf.signalClient.StopCatchSignals()
+			stackDescr, err := cf.cfnClient.Describe(stackName)
+			if err != nil {
+				return isDeleted, isUpdateCanceled, err
+			}
+			if aws.StringValue(stackDescr.StackStatus) == sdkcloudformation.StackStatusCreateInProgress {
```

Comment:
Would it be a better idea to use switch case for `aws.StringValue(stackDescr.StackStatus)`?